test/integration/resourcemanager/managedresource/resource_suite_test.go

```diff
@@ -123,8 +123,9 @@ var _ = BeforeSuite(func() {
 
 	Expect((&managedresource.Reconciler{
 		Config: config.ManagedResourceControllerConfig{
-			ConcurrentSyncs:     pointer.Int(5),
-			SyncPeriod:          &metav1.Duration{Duration: 500 * time.Millisecond},
+			ConcurrentSyncs: pointer.Int(5),
+			// Higher sync period is used since we don't want the changes in the test to be reconciled back quickly.
+			SyncPeriod:          &metav1.Duration{Duration: time.Minute},
```

Comment:
Why is this suddenly necessary? The comment is not quite helpful 👀 (at least I didn't understand the reason).

Reply:
In the test case, we need to delete, recreate it with different data and immutable, then reconcile the managedresource to assert that it is recreated. If the sync period is `500ms` immediately after deletion, the resource would be recreated.

Reply:
Sorry, that didn't help me to understand how this is related to this PR and why it wasn't causing problems before it.

Reply:
Only in this test, we are trying to assert an intermediate state. (The user tries to delete the secret and recreates it with their data and as immutable). If the controller acts every 500ms, then this would not be possible.
All other tests are fine since they expect an end state only.

Reply:
OK, thanks. Well, let's go ahead then.